File: turms-client-swift/Sources/TurmsClient/Model/MessageAddition.swift
public class MessageAddition {
    public let isMentioned: Bool
    public let mentionedUserIds: [Int64]
    public let recalledMessageIds: [Int64]

    init(isMentioned: Bool, mentionedUserIds: [Int64], recalledMessageIds: [Int64]) {
        self.isMentioned = isMentioned
        self.mentionedUserIds = mentionedUserIds
        self.recalledMessageIds = recalledMessageIds
    }
}

File: turms-client-dart/lib/src/model/message_addition.dart
import 'dart:core';

import 'package:fixnum/fixnum.dart' show Int64;

class MessageAddition {
  final bool isMentioned;
  final Set<Int64> mentionedUserIds;
  final Set<Int64> recalledMessageIds;

  MessageAddition(
      this.isMentioned, this.mentionedUserIds, this.recalledMessageIds);

  @override
  int get hashCode =>
      isMentioned.hashCode ^
      mentionedUserIds.hashCode ^
      recalledMessageIds.hashCode;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is MessageAddition &&
          isMentioned == other.isMentioned &&
          mentionedUserIds == other.mentionedUserIds &&
          recalledMessageIds == other.recalledMessageIds;

  @override
  String toString() =>
      'MessageAddition{isMentioned: $isMentioned, mentionedUserIds: $mentionedUserIds, recalledMessageIds: $recalledMessageIds}';
}
